pkg/cluster/backup_manager.go

```diff
@@ -154,7 +154,11 @@ func (bm *backupManager) makeSidecarDeployment() *appsv1beta1.Deployment {
 	case spec.BackupStorageTypeDefault, spec.BackupStorageTypePersistentVolume:
 		k8sutil.PodSpecWithPV(&podTemplate.Spec, cl.Metadata.Name)
 	case spec.BackupStorageTypeS3:
-		k8sutil.PodSpecWithS3(&podTemplate.Spec, c.S3Context)
+		if ss := cl.Spec.Backup.S3; ss != nil {
+			k8sutil.AttachS3ToPodSpec(&podTemplate.Spec, *ss)
```

Comment:
can we have a unified s3 configuration and function to handle how we attach s3 to a pod?